src/Matrix.cpp

```diff
@@ -0,0 +1,181 @@
+#include "Matrix.h"
+#include "ImageWrapper.h"
+
+
+//should probably convert this to a template
+
+matrix::matrix() : size(0), p_columns(0), p_rows(0), m_rows(0), m_columns(0), m_input_pixel_vector(0),
+m_size(0), m_ptr_double(nullptr) {}
+
+matrix::matrix(const uint8_t rows, const uint8_t columns, const std::vector<double> &greyscalevalues)
+{
+	m_rows = rows;
+	m_columns = columns;
+	m_size = rows * columns;
+	size = m_size;
+	p_columns = m_columns;
+	p_rows = m_rows;
+	m_greyscale_values = greyscalevalues;
+	
+	double* this_matrix = new double[m_size];
+	m_ptr_double = this_matrix;
+
+	for (int r = 0; r < rows; r++)
+	{
+		for (int c = 0; c < columns; c++)
+		{
+			this_matrix[c + r * columns] = greyscalevalues.at(c + r * columns);
+		}
+	}
+}
+
+/*
+//pixel vector version 
+matrix::matrix(const uint8_t rows, const uint8_t columns, const std::vector<pixel>& pixels)
+{
+	m_rows = rows;
+	m_columns = columns;
+	m_input_pixel_vector = pixels;
+	m_size = rows * columns;
+
+	pixel* this_matrix = new pixel[m_size];
+	m_ptr_pixel = this_matrix;
+
+	//fills matrix with [rows * columns] pixels, not a 2d array, still a regular 1d array.
+	//every r (row) increment jumps column (the width) elements forward, equivalent to moving down a row.
+	for (int r = 0; r < rows; r++)
+	{
+		for (int c = 0; c < columns; c++)
+		{
+			this_matrix[c + r * columns] = pixels.at(c + r * columns);
+		}
+	}
+
+	//prints pixels in matrix,		[R, G, B, A] [R, G, B, A] [R, G, B, A] [R, G, B, A]
+	//2x4 example, 8 pixels total	[R, G, B, A] [R, G, B, A] [R, G, B, A] [R, G, B, A]
+	for (int i = 0; i < rows * columns; i++)
+	{
+		std::cout << this_matrix[i];
+		if ((i + 1) % columns == 0)
+		{
+			std::cout << std::endl;
+		}
+	}
+}
+*/
+
+
+uint8_t matrix::get_row_amount() const
```

Comment:
if we already have these functions, why do we need p_columns, and p_rows?

Reply:
467934ac42a33b26e2cb83b6ccab2687a6ae9c28

same as above, public variables removed